libp2p/protocols/mix/entry_connection.nim

```diff
@@ -26,32 +39,126 @@ chronicles.formatIt(MixEntryConnection):
 method readOnce*(
     s: MixEntryConnection, pbytes: pointer, nbytes: int
 ): Future[int] {.async: (raises: [CancelledError, LPStreamError]), public.} =
-  # TODO: implement
-  raise newLPStreamEOFError()
+  if s.isEof:
+    raise newLPStreamEOFError()
+
+  try:
+    await s.replyReceivedFut
+    s.isEof = true
+    if s.cached.len == 0:
+      raise newLPStreamEOFError()
```

Comment:
i dont get this code, why us ` s.isEof = true` ?
it am confused here might be worth having a comment here.

Reply:
Fixed. This should happen only when cached is 0